src/client/entity-editor/relationship-editor/relationship.tsx

```diff
@@ -52,9 +53,8 @@ type RelationshipProps = {
 	relationshipType: RelationshipType
 };
 
-function Relationship({
-	contextEntity, link, relationshipType, sourceEntity, attributes, showAttributes, targetEntity
-}: RelationshipProps) {
+function Relationship(props: SingleValueProps<Relationship> | OptionProps<Relationship, any> | Relationship) {
+	const {contextEntity, link, relationshipType, sourceEntity, attributes, showAttributes, targetEntity} = props.data ?? props;
```

Comment:
It still looks wrong to me that props would be passed nested inside a `data` object.
We should avoid doing that, it's an anti-pattern.
Why is this necessary?